File: common/src/main/java/com/hartwig/actin/treatment/input/datamodel/TumorTypeInput.java
package com.hartwig.actin.treatment.input.datamodel;

import com.hartwig.actin.Displayable;

import org.jetbrains.annotations.NotNull;

public enum TumorTypeInput implements Displayable {
    CARCINOMA("305"),
    ADENOCARCINOMA("299"),
    SQUAMOUS_CELL_CARCINOMA("1749"),
    MELANOMA("1909");

    @NotNull
    private final String doid;

    TumorTypeInput(@NotNull final String doid) {
        this.doid = doid;
    }

    @NotNull
    public String doid() {
        return doid;
    }

    @Override
    @NotNull
    public String display() {
        return this.toString().replaceAll("_", " ").toLowerCase();
    }

    @NotNull
    public static TumorTypeInput fromString(@NotNull String string) {
        return TumorTypeInput.valueOf(string.trim().replaceAll(" ", "_").toUpperCase());
    }
}

File: common/src/main/kotlin/com/hartwig/actin/trial/input/datamodel/TumorTypeInput.kt
package com.hartwig.actin.trial.input.datamodel

import com.hartwig.actin.datamodel.Displayable

enum class TumorTypeInput(private val doid: String) : Displayable {
    CARCINOMA("305"),
    ADENOCARCINOMA("299"),
    SQUAMOUS_CELL_CARCINOMA("1749"),
    MELANOMA("1909");

    fun doid(): String {
        return doid
    }

    override fun display(): String {
        return this.toString().replace("_".toRegex(), " ").lowercase()
    }

    companion object {
        fun fromString(string: String): TumorTypeInput {
            return valueOf(string.trim { it <= ' ' }.replace(" ".toRegex(), "_").uppercase())
        }
    }
}
